packages/core/micro/src/consumerset.ts

```diff
@@ -0,0 +1,65 @@
+export type ConsumerSet<T> = undefined | T | Array<{ value: T, count: number }>
```

Comment:
Intended to provide efficient storage for a set of I*Consumers, but I'm dubious that it'll survive long.

I haven't benchmarked it against your much simpler `consumer0` solution, which is likely faster/simpler.

I'm also no longer convinced refcounting semantics are correct (mentioned in #58).

Reply:
> I haven't benchmarked it against your much simpler consumer0 solution, which is likely faster/simpler.

FWIW, I don't know if what I did was particularly good. I shameless ripped off https://github.com/janestreet/incremental

Reply:
Interesting!